src/io/perun.clj

```diff
@@ -391,6 +391,28 @@
     (require '~(symbol (namespace sym)))
     ((resolve '~sym) ~(pod/send! render-data))))
 
+(defn render-to-paths
+  [data renderer pod tmp]
+  (map
+   (fn [[path {:keys [render-data entry]}]]
+     (let [body (render-in-pod pod renderer render-data)]
+       (perun/create-file tmp path body)
+       (assoc entry :body body)))
+   data))
+
+(defn render-pre-wrap
+  [render-paths-fn options]
+  (let [pods (wrap-pool (pod/pod-pool (boot/get-env)))
+        tmp  (boot/tmp-dir!)]
+    (boot/with-pre-wrap fileset
+      (let [pod (pods fileset)
+            new-metadata (-> fileset
+                             (render-paths-fn options)
+                             (render-to-paths (:renderer options) pod tmp))]
+        (-> fileset
+            (perun/merge-meta new-metadata)
+            (commit tmp))))))
+
```

Comment:
I think it would be nice to add docstrings to those fns if we end up merging them :)